lib/client/index.js

```diff
@@ -52,74 +52,58 @@ module.exports = ({ port = null, config = {}, filters = {} }) => {
-    };
-
     // set auth header according to config
-    if (config.brokerClientValidationAuthorizationHeader) {
-      validationRequestHeaders.authorization =
-        config.brokerClientValidationAuthorizationHeader;
+    function credsFromHeader(s) {
+      if (s.indexOf(" ") >= 0) {
+        return s.substring(s.indexOf(" ") + 1);
+      } else {
+        return s;
+      }
+    }
+    let auths = [];
+    let rawCreds = [];
+    if (config.brokerClientValidationAuthorizationHeaderPool) {
+      auths = config.brokerClientValidationAuthorizationHeaderPool;
+      rawCreds = config.brokerClientValidationAuthorizationHeaderPool.map(credsFromHeader);
+    } else if (config.brokerClientValidationBasicAuthPool) {
+      auths = config.brokerClientValidationBasicAuthPool.map(s => `Basic ${Buffer.from(s).toString('base64')}`)
+      rawCreds = config.brokerClientValidationBasicAuthPool
+    } else if (config.brokerClientValidationAuthorizationHeader) {
+      auths.push(config.brokerClientValidationAuthorizationHeader);
+      rawCreds.push(config.brokerClientValidationAuthorizationHeader);
+      rawCreds = rawCreds.map(credsFromHeader);
     } else if (config.brokerClientValidationBasicAuth) {
-      validationRequestHeaders.authorization = `Basic ${Buffer.from(
-        config.brokerClientValidationBasicAuth,
-      ).toString('base64')}`;
+      auths.push(`Basic ${Buffer.from(config.brokerClientValidationBasicAuth).toString('base64')}`)
+      rawCreds.push(config.brokerClientValidationBasicAuth);
     }
 
     // make the internal validation request
-    request(
-      {
-        url: config.brokerClientValidationUrl,
-        headers: validationRequestHeaders,
-        method: brokerClientValidationMethod,
-        timeout: brokerClientValidationTimeoutMs,
-        json: isJsonResponse,
-        agentOptions: {
-          ca: config.caCert, // Optional CA cert
-        },
-      },
-      (error, response) => {
-        // test logic requires to surface internal data
-        // which is best not exposed in production
-        if (process.env.TAP) {
-          data.testError = error;
-          data.testResponse = response;
-        }
-
-        if (error) {
-          data.ok = false;
-          data.error = error.message;
-          return res.status(500).json(data);
-        }
-
-        const responseStatusCode = response && response.statusCode;
-        data.brokerClientValidationUrlStatusCode = responseStatusCode;
-
-        // check for 2xx status code
-        const goodStatusCode = /^2/.test(responseStatusCode);
-        if (!goodStatusCode) {
-          data.ok = false;
-          data.error =
-            responseStatusCode === 401 || responseStatusCode === 403
-              ? 'Failed due to invalid credentials'
-              : 'Status code is not 2xx';
-
-          logger.error(data, response && response.body, 'Systemcheck failed');
-          return res.status(500).json(data);
-        }
+    const rv = [];
```

Comment:
`rv` = request validated?
Maybe there is more verbose argument name

Reply:
`returnValue`, and yeah, you're right.